tests/e2e/runtime.go

```diff
@@ -0,0 +1,465 @@
+
+		sc.Logger.Info("running test", "test", testName)
+		if testErr := test(sc.Net, rtc); testErr != nil {
+			sc.Logger.Error("test failed",
+				"test", testName,
+				"err", testErr,
+			)
+			return testErr
+		}
+		sc.Logger.Info("test passed", "test", testName)
+	}
+
+	return sc.Net.CheckLogWatchers()
+}
+
+// Ugh, this is really annoying :/
+//
+// cannot use sw.signer.Public() (type "github.com/oasisprotocol/oasis-core/go/common/crypto/signature".PublicKey) as type "github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature".PublicKey in return argument:
+//	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature".PublicKey does not implement "github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature".PublicKey (wrong type for Equal method)
+//		have Equal("github.com/oasisprotocol/oasis-core/go/common/crypto/signature".PublicKey) bool
+//		want Equal("github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature".PublicKey) bool
+type pkWrapper struct {
+	pk coreSig.PublicKey
+}
+
+func newPublicKeyWrapper(pk coreSig.PublicKey) *pkWrapper {
+	return &pkWrapper{pk: pk}
+}
+
+func (pk *pkWrapper) String() string {
+	return pk.pk.String()
+}
+
+func (pk *pkWrapper) Equal(other signature.PublicKey) bool {
+	// Can't compare bytes, because signature.PublicKey can't be sliced...
+	return pk.String() == other.String()
+}
+
+func (pk *pkWrapper) Verify(context, message, signature []byte) bool {
+	return pk.pk.Verify(coreSig.Context(context), message, signature)
+}
+
+// Also really annoying:
+//
+// cannot use signer (type "github.com/oasisprotocol/oasis-core/go/common/crypto/signature".Signer) as type "github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature".Signer in argument to kvInsert:
+//	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature".Signer does not implement "github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature".Signer (wrong type for ContextSign method)
+//		have ContextSign("github.com/oasisprotocol/oasis-core/go/common/crypto/signature".Context, []byte) ([]byte, error)
+//		want ContextSign([]byte, []byte) ([]byte, error)
+type signerWrapper struct {
+	signer coreSig.Signer
+}
+
+func newSignerWrapper(signer coreSig.Signer) *signerWrapper {
+	return &signerWrapper{signer: signer}
+}
+
+func (sw *signerWrapper) Public() signature.PublicKey {
+	return newPublicKeyWrapper(sw.signer.Public())
+}
+
+func (sw *signerWrapper) ContextSign(context, message []byte) ([]byte, error) {
+	return sw.signer.ContextSign(coreSig.Context(context), message)
+}
+
+func (sw *signerWrapper) String() string {
+	return sw.signer.String()
+}
+
+func (sw *signerWrapper) Reset() {
+	sw.signer.Reset()
+}
+
+type kvKey struct {
+	Key string `json:"key"`
+}
+
+type kvKeyValue struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+func kvInsert(rtc client.RuntimeClient, signer signature.Signer, nonce uint64, key string, value string) error {
```

Comment:
Consider moving these key-value specific APIs into a separate package that you then use in tests.